pkg/kubernetes/controller.go

```diff
@@ -1,12 +1,162 @@
 package kubernetes
 
 import (
+	"bytes"
 	"context"
+	"errors"
+	"fmt"
+	"time"
 
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/remotecommand"
 )
 
 func (k *Kubernetes) GetPodStatus(namespace, name string) (string, error) {
 	pod, err := k.client.CoreV1().Pods(namespace).Get(context.TODO(), name, v1.GetOptions{})
 	return string(pod.Status.Phase), err
 }
+
+func (k *Kubernetes) CreateConfigMap(namespace string, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+	return k.client.CoreV1().ConfigMaps(namespace).Create(context.TODO(), configMap, v1.CreateOptions{})
+}
+
+func (k *Kubernetes) UpdateConfigMap(namespace string, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+	return k.client.CoreV1().ConfigMaps(namespace).Update(context.TODO(), configMap, v1.UpdateOptions{})
+}
+
+func (k *Kubernetes) GetConfigMap(namespace string, name string) (*corev1.ConfigMap, error) {
+	return k.client.CoreV1().ConfigMaps(namespace).Get(context.TODO(), name, v1.GetOptions{})
+}
+
+func (k *Kubernetes) CreateDeployment(namespace string, deployment *appsv1.Deployment) (*appsv1.Deployment, error) {
+	return k.client.AppsV1().Deployments(namespace).Create(context.TODO(), deployment, v1.CreateOptions{})
+}
+
+func (k *Kubernetes) DeleteDeployment(namespace string, name string) error {
+	return k.client.AppsV1().Deployments(namespace).Delete(context.TODO(), name, v1.DeleteOptions{})
+}
+
+func (k *Kubernetes) CreateNamespace(name string) (*corev1.Namespace, error) {
+	namespace := &corev1.Namespace{
```

Comment:
@rlgns98kr I tried ti use only k.client.CoreV1().Namespaces().Create(context.TODO(), namespace, v1.CreateOptions{}), but it seems make us harder for maintainer or unittest since it is quite complex for creating mock object. What do you think?

Reply:
Oh, I understood. It is too long to use. Okay, let's keep this code. It's okay to me. I'm sorry for confusion